src/modules/common/helpers/response-filters.ts

```diff
@@ -38,4 +38,56 @@ export const chapterListResponseFilter = (apiResponse: IChaptersAPIResponse) =>
         return chapterListItem
     })
     return chapterList
+}
+
+export const carouselFilter = (apiResponse: ITopComicsAPIResponse, category: string) => {
+    switch (category) {
+        case "new":
+            {
+                const newMangaListState: ITrendingMangaApp[] = apiResponse.news.map((manga: INews) => {
+                    return {
+                        slug: manga.slug,
+                        title: manga.title,
+                        coverImage: useMangaStore.getState().imageUrlPrefix + manga.md_covers[0].b2key,
```

Comment:
Remember to do your checks for api response data, not checking for the existence of these might throw errors

Reply:
Thanks for the heads up, bro! Safety checks have been included now!